From 52b7af9e57b16a940edec5830f7458891f082221 Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Wed, 18 Jan 2023 14:34:20 +0100
Subject: [PATCH 249/954] feat(stm32mp2): manage PKH OTP

The hash of public keys hashes is stored in 8 OTP fuses,
from 144 to 152.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 611c1c1c29bc3694943530b0235a8d5b66ed2a09 feat(stm32mp2): manage PKH OTP]
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Change-Id: I07e5424c80d5bd3a638ccdacfc1c658c3c2e5c45
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/285744
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Lionel DEBIEVE <lionel.debieve@foss.st.com>
---
 fdts/stm32mp251.dtsi            | 3 +++
 plat/st/stm32mp2/stm32mp2_def.h | 1 +
 2 files changed, 4 insertions(+)

diff --git a/fdts/stm32mp251.dtsi b/fdts/stm32mp251.dtsi
index 50a3ba5bb3..87891a6440 100644
--- a/fdts/stm32mp251.dtsi
+++ b/fdts/stm32mp251.dtsi
@@ -314,6 +314,9 @@
 			hconf1_otp: otp124@1f0 {
 				reg = <0x1f0 0x4>;
 			};
+			pkh_otp: otp144@240 {
+				reg = <0x240 0x20>;
+			};
 		};
 
 		iwdg1: watchdog@44010000 {
diff --git a/plat/st/stm32mp2/stm32mp2_def.h b/plat/st/stm32mp2/stm32mp2_def.h
index cf6b2b2772..bcad5938d3 100644
--- a/plat/st/stm32mp2/stm32mp2_def.h
+++ b/plat/st/stm32mp2/stm32mp2_def.h
@@ -315,6 +315,7 @@ enum ddr_type {
 #define BOARD_ID_OTP				"board_id"
 #define UID_OTP					"uid_otp"
 #define LIFECYCLE2_OTP				"otp18"
+#define PKH_OTP					"otp144"
 
 /* OTP mask */
 /* PACKAGE */
-- 
2.53.0

